### DISCUSSION TITLE
Status auf off

### DISCUSSION THREAD

**Discussion by woupi** [camueller/SmartApplianceEnabler#261] [Allgemein]
Guten Morgen, 

bis Samstag lief meine Anlage mit insgesamt 10 modbus Zähler incl. SunnyPortal einwandfrei.

Da ich die Anlage um weiter Modbus Zähler, einen Shelly 4pro Erweitert und Edimax Steckdose dafür entfernt habe, musste ich nun an die Software Seite. Das versuche ich nun seit Samstag Abend leider Erfolglos.

Mein Aufbau sieht und sah auch vorher so aus:
Proxmox auf Nuc, SAE in LXC und iobroker in VM. Die Modbus Zähler werden vom iobroker Adapter erfasst und mit simpelAPI als http bereitgestellt (Bild einer Abfrage hängt unten an).
SAE als Typ http konfiguriert, genauso wie vorher auch, die alten sind ja auch genauso geblieben, da habe ich nichts dran verändert, nur neu hinzugefügt.

Anschließend habe ich alle Geräte/Verbraucher die von SAE gekommen sind bzw. die Edimax Steckdosen im SunnyPortal gelöscht.

Dann neue Geräte gesucht und anschließend alle auf einmal ins SunnyPortal implementiert. Nachdem alle Geräte da waren noch im SunnyPortal umbenannt und fertig, habe ich gedacht. War aber nichts.
Denn es kommen keine Werte, auch nicht von denen, die vorher einwandfrei funktioniert haben.

Ich habe als Bilder mal ein Zähler im SAE angehangen, das Gerät im semp Protokol und die dazu gehöhrende http Abfrage.

Ich bin leider kit meinem Latein am Ende und hoffe hier auf Hilfe.

Danke schon mal im Voraus und LG

SAE <DeviceId>F-23051969-000000000002-00</DeviceId> 

![SAE](https://user-images.githubusercontent.com/98451301/166191497-451198a5-60d5-45c2-a5d2-e55a058cf0fd.jpg)

SEMP

![SAE_semp](https://user-images.githubusercontent.com/98451301/166191504-4ef3488c-67d7-46be-a90b-52cf83026bfa.jpg)

http

![http_abfrage](https://user-images.githubusercontent.com/98451301/166191514-aa16e42b-fa72-464c-9ed2-eb0f51fafdad.jpg)

**Comment by camueller** [comment]
Wenn man ein einwandfrei laufendes System hat, verstehe ich nicht, warum man dann alles löscht. Selbst wenn man an der Grenze von 12 Geräte ist, löscht man nur soviele Geräte, bis man unter der Grenze ist und fügt dann die neuen Geräte hinzu. Aber **alle** Geräte löschen? Und warum die SAE-Konfigurationen der Geräte löschen? Und wenn man die SAE-Konfigurationen löscht, dann sichert man vorher die XML-Files, damit man immer wieder zurück kann auf den bekanntermassen funktionierenden Stand.

Konkrete Hinweise kann ich Dir leider nicht geben ... Vielleicht mal den SHM resetten !?

**Comment by woupi** [comment]
Du hast mich falsch verstanden, ich habe nicht alles gelöscht sondern, alles was über den SAE kommt und alle Steckdosen die ich nicht mehr nutzen möchte.

Den Weg bin ich gegangen, damit ich an die max. Grenze von 22 Geräten komme.

Die Konfiguration im SAE habe ich bei den bestehenden Geräten übrigens nicht gelöscht, habe ich oben aber auch geschrieben  (die alten sind ja auch genauso geblieben, da habe ich nichts dran verändert, nur neu hinzugefügt.).

SHM hatte ich auch schon resetet.

Ich bin gerade auf Geschäftsreise und habe daher keinen Zugriff auf SAE und semp, aber ich habe gerade im Hotel nochmal alles was von SAE kommt im SunnyPortal gelöscht und nochmal alles neu hinzugefügt.
Jetzt werden mir bei einigen auch wieder Daten angezeigt. Nur verifizieren kann ich das von hier nicht, dafür müsste ich im Zählerschrank auf die Zähler schauen.

Was auch immer da falsch gelaufen ist, ich hoffe jetzt einfach mal, dass es jetzt wieder geht.

LG

**Comment by woupi** [comment]
Guten Morgen,

ich habe mich zu früh gefreut, offensichtlich kommen einige Werte doch nicht an.
Wirklich dabei sehen kann ich erst am Wochenende, wenn ich wieder zu Hause bin, aber eine Frage hätte ich schon mal.
Ich habe in den FAQ zwar gefunden das der Status auf on stehen muss aber nicht was ich machen kann wenn er eben auf off steht.
Ich kann mich an einen Zähler im semp erinnern, der einen Wert (AvergagePower) angegeben hatte (meine 11) aber unter Status auf off stand.
LG

**Reply by premultiply** [reply]
Zeitplan angelegt?
Schaltertyp auf „Immer eingeschaltet“ oder verbrauchsbasiert gestellt?

**Comment by woupi** [comment]
Bei dem bestimmten, den ich gesehen habe, handelt es sich um den Zähler der Wärmepumpe. Es ist kein Schalter konfiguriert und dem entsprechend kein Zeitplan und kein Schaltertyp.

**Reply by premultiply** [reply]
Dann Schaltertyp Immer eingeschaltet oder Verbrauchsbasiert.
Ohne geht es nicht.

Geräte brauchen bei SEMP einen Status. Der kommt über den Schalter.
Geräte die ausgeschaltet sind werden ignoriert.

**Comment by woupi** [comment]
Ah ok, vielen Dank, das werde ich sofort am Freitag, wenn ich nach Hause komme ausprobieren.

Hast Du eine Erklärung dafür, warum dieser Zähler, der ist unverändert geblieben, bis Samstag einwandfrei funktioniert hat und erst jetzt, nachdem ich viele neue hinzugefügt habe, diese Problem macht? Einen Schalter habe ich da noch nie eingestellt.

Würde es gerne verstehen, einfach damit ich, oder aber auch andere, die das selbe Problem haben, den Fehler eben nicht machen bzw nochmal machen.

**Comment by woupi** [comment]
Ich habe gerade in den Grundlagen den Hinweis gesehen, das der Schalter immer auf eingeschaltet stehen muss. Gelesen habe ich das bestimmt, wie ich vor einem Jahr damit angefangen habe, aber bei alle dem was hier in dem Jahr passiert ist, habe ich das schlicht vergessen.
Von daher kann ich jetzt auch nicht mehr mit Bestimmtheit sagen, ob ich nicht doch bei der WP den Schalter auf eingeschaltet gestellt habe, erinnern kann ich mich nicht daran. Aber dann dürfte sich da eigentlich nichts dran geändert haben.
Ist wie es ist, ich kann leider erst am WE dabei sehen.

**Comment by woupi** [comment]
Guten Morgen,

ich habe mich gestern, nachdem ich wieder zu Hause war, direkt daran gegeben.
Ich habe alle Schalter auf "immer eingeschaltet" gestellt. Sie haben auf "Zustandsbasierter Zustandsmelder" gestanden.
Bei manchen kann man das allerdings nicht einstellen, da es schlicht diesen Punkt gar nicht gibt und zwar bei Wallbox und bei dem Shelly.

Jetzt sind mir erstens die Woche über schon einige Dinge und auch seit gestern Abend aufgefallen, die sehr merkwürdig sind.

1. Die Wärmepumpe. Sie wurde in SAE nicht angefasst, also nichts verändert. Im SEMP steht sie auf "on" und es wird ein Wert von "11" übergeben. 
Im SunnyPortal steht unter Tag bei Verbraucherbilanz und -steuerung allerdings für die Wärmepumpe nichts. 
Unter "Monat" steht an manchen tagen allerdings manchmal etwas und an manchen nichts, gelaufen ist sie aber jeden Tag und die Werte die an den Tagen, an denen etwas im SunnyPortal steht, stehen, passen nicht mit den Werten überein, die die Wärmepumpe tatsächlich gebraucht hat.
Ich habe gestern z.B. extra die Wärmepumpe auf Warmwasser gestellt, damit sie auch auf jeden Fall läuft.
Tatsächlicher Verbrauch am Zähler direkt abgelesen 1437 Watt, im SEMP 1437 Watt, unter "aktuell" im Sunny Portal 1437 Watt. Passt also alles. Das ganze habe ich dann ca. 1/2 stunde laufen lassen und immer mal wieder nach gesehen, mehr oder weniger passten die Werte zusammen.
Aber im Sunny Portal unter Tag  steht nichts und auch jetzt, mehrere Stunden später steht unter gestern zur Wärmepumpe nichts.

Das war jetzt nur ein Beispiel, gilt aber für fast alle Verbraucher, unter dem Tag werden nur ganz wenige, von den 22 Verbrauchern die übermittelt werden angezeigt. Bei dem Aktuellen ist es Glückssache ob etwas angezeigt wird oder nicht, ich habe mehrere Zähler verglichen.

2. Bei der Wallbox, bei der es den Punkt "Schalter" im SAE nicht gibt, steht im SEMP unter Status: Off.
In der Woche hat meine Frau mehrmals das Auto an der Wallbox gehabt, im Sunny Portal steht davon aber nichts, obwohl ich im SAE auch hier gar nichts geändert habe.
Ob vor hinzufügen der anderen Zähler im SAE unter SEMP bei der Wallbox im Status: On gestanden hat, kann ich nicht sagen. da zur der Zeit alles super funktioniert hat, gab es gar keinen Grund im SEMP nachzusehen.

Ist es möglich, dass das SunnyPortal mit 22 Zählern überfordert ist, anders kann ich es mir gerade nicht erklären, oder hat einer eine andere Idee?

LG

**Comment by camueller** [comment]
Eine Wallbox ist ein Gerätetyp mit einem Schalter mit variabler Leistungsaufnahme, insofern kann man da keinen weiteren Schaltertyp angeben. Für alle anderen Gerätetypen kann man den Schaltertyp angeben. Wenn der Shelly als Zähler eingesetzt wird ohne dass ein Schalter konfiguriert wird, ist implizit automatisch der Schalter auf "Zustandsbasierter Zustandsmelder" gesetzt. Das steht alles [in der Doku](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Control_DE.md),

Wenn (wie bei Deiner Wallbox) erwartete Leistungsaufnahmen aus den vergangenen Tagen nicht im Sunny Portal auftauchen hilft ein Blick in die SAE-Logifiles. Für jede Abfrage des SAE durch den SHM findet sich ein Eintrag im Log, in welchem die gemeldete Leistungsaufnahme enthalten ist - hier sind es 4108W von meiner Wärmepumpe:
```
2022-05-04 04:34:55,638 DEBUG [http-nio-8080-exec-7] d.a.s.s.w.SempController [SempController.java:323] F-00000001-000000000001-00: Reporting power info from meter.
2022-05-04 04:34:55,640 DEBUG [http-nio-8080-exec-7] d.a.s.s.w.SempController [SempController.java:337] F-00000001-000000000001-00: PowerInfo{averagePower=4108, minPower=null, maxPower=null, timestamp=0, averagingInterval=60}
```

Die Frage, ob das Sunny Portal mit 22 Zählern überfordert ist, solltest Du erst stellen, wenn Du **sicher** bist, dass bis zur SEMP-Schnittstelle auf Deiner Seite alles korrekt ist.

**Comment by woupi** [comment]
Zum Shelly, Du hast natürlich recht, habe ich auch gemacht und zwar auf http, sorry.
Ändert allerdings nichts daran, das obwohl er eingeschaltet ist, Stus auf off steht.
Dabei gehe ich allerdings davon aus, das ich bei der abfrage noch einen Fehler habe, darum hier ein screenshot:

![image](https://user-images.githubusercontent.com/98451301/167246967-b5c2d4cf-97cb-4fce-a4c6-745ea8594986.png)

So sieht die abfrage über http aus:
{"ison": true, "has_timer":false, "timer_started_at": 0, "timer_duration": 0.00, "timer_remaining": 0.00, "overpower": false, "source": "WS_in"}

Mit den logs muss ich mich morgen beschäftigen, muss jetzt leider arbeiten, sorry und Danke für deine Hilfe.

mfg

**Comment by woupi** [comment]
Hallo Axel,

ich habe mir jetzt gerade die log Files zu dem Shelly Ph 1 angesehen. Da dieser heute genau das gemacht hat, was ich meine.
Er wurde über SunnyPortal und SAE geschaltet, das hat also schonmal funktioniert, wenn auch nicht so wie ich mir das vorstelle, aber das zu einem späteren Zeitpunkt.

Im Semp werden die Daten übermittelt und im Sunny Portal unter  --> "Aktuell" wurden auch die Werte aus dem Semp angezeigt.

Jetzt einige Stunden später sieht man allerdings unter "Verbraucherbilanz und -steuerung" --> "Tag" von dem Shelly Ph 1 nichts, nicht ein Watt und er lief einige Stunden.

Wenn ich jetzt in den log schaue, steht dort z.B., wobei dazu gesagt werden muss, dass die Ph 1 zwar geschaltete ist, aber da es sich um einen Heizstab handelt und der Temperaturschalter abgeschaltet hat, fließt jetzt gerade kein Strom mehr.

`2022-05-07 13:32:04,029 DEBUG [Timer-0] d.a.s.h.HttpHandler [HttpHandler.java:92] F-23051969-000000000014-00: Response: {"id":1, "source":"http", "output":true, "apower":0.0, "voltage":235.3, "current":0.018, "pf":0.00, "aenergy":{"total":0.000,"by_minute":[0.000,0.000,0.000],"minute_ts":1651930322},"temperature":{"tC":36.4, "tF":97.5}}`

Was mich halt wundert, dass unter "Aktuell" im SunnyPortal etwas angezeigt wurde und in der Tages Ansicht davon nichts zu sehen ist.

Und so zieht sich das leider durch, wobei manchen Werte teilweise angezeigt werden, aber eben nicht alles. Es war z.B. an einem Tag so, dass die Wärmepumpe über den Tag 1045Wh laut Zähler verbraucht hat, aber im SunnyPortal nur 437Wh gestanden haben.

Warum wird manchmal gezählt und manchmal nicht.

**Reply by camueller** [reply]
Wenn das Sunny Portal die richtigen Werte unter "Aktuell" anzeigt, hat der SAE alles richtig gemacht und der SHM die Daten empfangen. Wenn das Verhalten weiterhin so ist, bleibt Dir nichts weiter übrig, als die an SMA zu wenden. Bitte berichte hier, wie es in dieser Sache weitergeht.

**Comment by woupi** [comment]
Guten Morgen,

ok, ich werde es weiter beobachten und ggf eine Anfrage an SMA stellen.

Drei Auffälligkeiten habe ich allerdings noch zu dem Verhalten vom SAE bei dem Shelly festgestellt:

1.  Der Shelly wurde von SMA über SAE eingeschaltet und zwar auch Phasenweise (ist der 4 pro). Im Semp hat auch der gemessene Wert gestanden und trotzdem stand unter Status Off. 
2. Im SAE unter Status bleibt die Ampel dabei auf gelb.
3. Die Ampel schaltet auch nicht auf grün, obwohl ich auf die grüne Ampel geklickt habe und anschließend auf Start.

Zum weiteren Testen habe ich die Schaltzeiten nochmal geändert, so steht jetzt die Ampel auf rot und ich kann nachher nochmal versuchen eine Phase über die Ampel zu schalten. Mal sehen was dann mit der Ampel passiert und was dann unter Status beim SEMP steht.

**Comment by camueller** [comment]
Der SAE bietet die Möglichkeit, Schalter und Zähler über verschiedene Protokolle anzubinden. Beim Shelly ist das HTTP. Da ich selbst keine Shelly-Produkte verwende, kann ich nicht untertstützen, wenn diese nicht erwartungsgemäss funktionieren. Evtl. ist die Dokumentation für die Shelly veraltet oder falsch. **Hier sind die SAE-User von Shelly-Produkten gefordert, diese aktuell zu halten.**

**Comment by woupi** [comment]
Danke trotzdem Axel.
Eventuell findet sich ja ein Shelly Benutzer der ähnliche Festgestellt hat und evt. auch ein Lösung hat.
Ich werde weiter Experimentieren und falls ich den richtigen weg der Einstellungen gefunden habe mich melden.

In die Programmierung gehe ich nicht, da ich 1. weit davon weg bin und 2. leider auch immer noch keine Zeit habe mich um so nebenkriegsplätze zu kümmern, erst muss mein Haus fertig sein. 
Ich rechne z.Z. damit, dass das noch ca. 2 Jahre dauern wird. 
Wenn bis dahin das Problem nicht gelöst ist, würde ich mich dann ggf intensiv damit auseinandersetzen.

Danke trotzdem wie immer für deine Arbeit und deine Unterstützung

**Comment by woupi** [comment]
Ich habe jetzt gerade über die Ampel eine Phase des Shelly eingeschaltet, Ampel springt von Rot auf Gelb. Im SEMP steht weiterhin bei Status Off und die Leistung wird angezeigt.

Da stellen sich mir 2 Fragen:

1. Ist es normal, dass in diesem Szenario die Ampel nicht auf Grün springt und
2. Müsste nicht eigentlich der Status auf On springen, oder springt der nur auf On, wenn die Ampel auch grün zeigt?

Axel, da zumindest die 2. Frage ja nichts mit dem Shelly zu tun hat, könntest du diese bestimmt beantworten.

Danke

LG

**Reply by Trunks1982** [reply]
Zeig doch mal deinen shelly befehl!

**Reply by woupi** [reply]
Gerne, wobei weiter ob die schon stehen, hier jetzt dann mal alle, also Haupt-, Zähler-, Schalter- und Schaltzeiten-Seite von einer Phase, die anderen sehen dann bis auf die Nummerung gleich aus:

![image](https://user-images.githubusercontent.com/98451301/167301989-a4e27a03-aa8e-4325-9925-44c1491b11b9.png)

Zähler:
![image](https://user-images.githubusercontent.com/98451301/167302023-163fa5b8-bf47-4e0c-8fc6-deaeb477ee19.png)

Schalter:
![image](https://user-images.githubusercontent.com/98451301/167302060-ef5e38bf-7074-4de9-98ba-343c1a7ebf19.png)

Schaltzeiten:
![image](https://user-images.githubusercontent.com/98451301/167302092-75c0f027-15b6-47d3-b9e7-e40345a1f397.png)

**Reply by Trunks1982** [reply]
Was willst du mit der Mindesteinschaltzeit von 1 sek bezwecken?Reiner Überschuss ist zb 00:00

**Reply by woupi** [reply]
00:00 hatte ich meine ich versucht

**Reply by woupi** [reply]
ok, habe es gerade nochmal probiert, hat funktioniert. Weiß nicht war es gestern nicht funktioniert hat.

Und, ja am liebsten hätte ich die Heizstäbe nur dann, wenn ich über die 70% bei der Einspeisung komme.

**Reply by Trunks1982** [reply]
![Bildschirmfoto 2022-05-08 um 17 10 36](https://user-images.githubusercontent.com/62701386/167302680-fdf7cd25-ed98-4fc3-8d29-6e0fd3583150.png)
Das funktioniert Wunderbar.Reine Überschusssteuerung.Die Shelly befehle scheinen richtig zusein.Statusabfrage würde ich zb rausnehmen.Brauchste eigentlich nicht.

**Reply by woupi** [reply]
Das habe ich reingemacht, da die Ampel nicht wie erwartet funktioniert hat und ich ausprobieren musste.

**Reply by Trunks1982** [reply]
Nimm mal raus und guck mal ob du dann über die Ampel Ansteuern kannst

**Reply by woupi** [reply]
Funktioniert das bei dir mit deinem Zeitplan so, dass er nur einschaltet wenn Du über 70% Einspeisung kommt oder immer wenn Du mehr produzierst als im Haus bleibt?

**Reply by woupi** [reply]
Super, dass war es was die Ampel betrifft und im SEMP steht jetzt auch Status auf on.
Danke

**Reply by Trunks1982** [reply]
Hab es für immer eingestellt.Du kannst doch im sunnyportal die reglung dür den schalter einstellen.

**Reply by camueller** [reply]
Schön, dass es jetzt funktioniert! Ich kann aber keine Unterschied zwischen Deinem Screenshot und der Doku erkennen.

**Reply by woupi** [reply]
"Du kannst doch im sunnyportal die reglung dür den schalter einstellen."

Ok, habe den Wald vor lauter Bäumen nicht gesehen, Du hast natürlich recht, werde ich dort jetzt mal anpassen und schauen ob es so funktioniert wie ich mir das vorstelle.

Vielen Dank für deine Hilfe, das hat mir mit Sicherheit viele Stunden gespart.

"Ich kann aber keine Unterschied zwischen Deinem Screenshot und der Doku erkennen."
Ich habe auch keinen Unterschied gefunden und da es jetzt funktioniert, kann die Doku auch genau so bleiben, ich wüsste zumindest nicht, was Du da anders hättest schreiben können.

Auch Dir nochmal vielen Dank.

LG

**Reply by woupi** [reply]
Ich habe jetzt alle im Sunny Portal auf 100% Überschuss eingestellt und die Zeitpläne auch von 8 bis 17 Uhr und auf max 5h gestellt.
Jetzt blinkt auch die Ampel gelb, so wie in der Beschreibung.

Super, wieder ein Stückchen weiter.

Vielen lieben Dank, jetzt kann ich mich wieder den anderen Baustellen widmen.

LG

**Comment by camueller** [comment]
Wenn man druch Kick auf das unterste Ampellicht das Gerät einschaltet, sollte danach das grüne Licht auch leuchten. Wenn das nicht so ist, ergibt die Schalter-Abfrage des Gerätes nicht, dass es sich im Status eingeschaltet befindet. Insofern stimmt Deine Shelly-Ansteuerung immer noch nicht. Die Ampel visualisiert nur den Zustand des Gerätes, wobei man durch Klicks auf die Ampel das Gerät auch steuern kann (was normalerweise der SHM tut).

**Reply by woupi** [reply]
Das mit dem Klick und dann die Änderung der Farbe auf den jeweiligen Zustand, hatte ich auch so erwartet. Ok, wenn das bei mir nicht so ist, muss ich suchen, was ich da falsch habe.

Wobei ich alle Befehle, also z.B. http://192.168.1.161/relay/1?status, durch aufrufen mit dem Browser, schon ausprobiert habe. Sie funktionieren alle. Ich kann also einschalten und bekomme über den status Befehl als Rückantwort:
{"ison": true, "has_timer":false, "timer_started_at": 0, "timer_duration": 0.00, "timer_remaining": 0.00, "overpower": false, "source": "http"}.

Von daher, weiß ich leider nicht mehr wo ich noch was falsch gemacht haben könnte, aber meistens liegt der Fehler ja im Detail und so kann ich nur hoffen, dass ich ihn irgendwann oder einer in findet.

LG
